prow/github/client.go

```diff
@@ -986,7 +988,7 @@ func (c *client) requestRaw(r *request) (int, []byte, error) {
 }
 
 func (c *client) requestRawWithContext(ctx context.Context, r *request) (int, []byte, error) {
-	if c.fake || (c.dry && r.method != http.MethodGet) {
```

Comment:
/cc @petr-muller @chaodaiG 

WDYT? I understand the thought here but it is also going to be surprising

Reply:
I'm leaning towards allowing it. Authentication is special and is usually not thought as being a write operation, which is my bar for what I expect from dry runs. It has legit use cases. 

Technically it *is* a write operation though - are we sure there is no observable side effects of executing such dry-run other than possibly obtaining a new installation token that will expire anyway?

Reply:
On the exact same page with @petr-muller , imo it's fine to allow requesting this token as part of dry-run. Might be helpful to make it a bit more clear on the allowed request path, with one of the two approaches:

a).  Commenting the regex to make it clear why
b). Encapsulate the condition in a function called `isRequestingAppInstallationToken`

On the other hand, adding one more flag `--dryrun-allow-request-app-installation-token` to make this configurable could be helpful to avoid surprising side effect. We can add an announcement at https://github.com/kubernetes/test-infra/blob/master/prow/ANNOUNCEMENTS.md#new-features and stating that this feature will become default in 6 months etc.

Reply:
sgtm, while it is a mutating request it doesn't result in any observable state change on GitHub